source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using AutoFixture.Xunit2;
+using Energinet.DataHub.Wholesale.Application.Base;
+using FluentAssertions;
+using MediatR;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.WebApi.IntegrationTests.Application;
+
+public class RequestMatchingHandlerTests
+{
+    /// <summary>
+    /// A request can be either a command or a query.
+    /// </summary>
+    [Theory]
+    [InlineAutoData(typeof(Root))]
+    [InlineAutoData(typeof(Wholesale.Application.Root))]
+    [InlineAutoData(typeof(Domain.Root))]
+    [InlineAutoData(typeof(Wholesale.Infrastructure.Root))]
+    public void EachRequest_HasMatchingHandler(Type type)
```

Comment:
What are your thoughts on this suggestion?
```suggestion
    public void EachCommand_HasASingleHandler(Type type)
```